src/aktualizr_follower/follower_config.cc

```diff
@@ -0,0 +1,44 @@
+#include <sstream>
+
+#include "config_utils.h"
+#include "follower_config.h"
+
+FollowerConfig::FollowerConfig(const boost::filesystem::path& filename,
+                               const boost::program_options::variables_map& cmd) {
+  updateFromToml(filename);
+  updateFromCommandLine(cmd);
+}
+
+FollowerConfig::FollowerConfig(const boost::filesystem::path& filename) { updateFromToml(filename); }
+
+void FollowerConfig::updateFromCommandLine(const boost::program_options::variables_map& cmd) {
+  if (cmd.count("server-port") != 0) {
+    network.port = cmd["server-port"].as<int>();
+  }
+}
+
+void FollowerConfig::updateFromPropertyTree(const boost::property_tree::ptree& pt) {
+  // Keep this order the same as in follower_config.h and writeToFile().
+  CopyFromConfig(network.port, "network.port", boost::log::trivial::info, pt);
```

Comment:
Shouldn't we just accept the socket systemd gives us?

Reply:
That's if we don't want to be tied to systemd in all cases, eg for testing.
But that should not be hard to add it back when we'll need it, so yes we could omit this part.